test.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * test.c
+ *
+ *  Created on: Feb 20, 2020
+ *      Author: Youssef Harby
+ */
+
+
+#include "DIO_TS/DIO.h"
+#include <stdio.h>
+
+DIO_Cfg_s cfg_test;
+
+void DIO_initTesting(uint8_t gpio, uint8_t bit, uint8_t dir, uint8_t expected_output)
+{
+	static int initTestCounter = 1;
+	uint8_t output;
+
+	cfg_test.GPIO = gpio;
+	cfg_test.pins = bit;
+	cfg_test.dir = dir;
+
+	output = DIO_init(&cfg_test);
+
+	if(output == expected_output)
+	{
+		printf("DIO_init test %d Passed\n", initTestCounter);
+		fflush(stdout);
+	}else
+	{
+		printf("DIO_init test %d failed\n", initTestCounter);
+		fflush(stdout);
+	}
+
+	initTestCounter++;
+}
+
+void DIO_writeTesting(uint8_t gpio, uint8_t bits, uint8_t value, uint8_t expected_output)
+{
+	static int writeTestCounter = 1;
+	uint8_t output;
+
+	output = DIO_Write(gpio, bits, value);
+
+	if(output == expected_output)
+	{
+		printf("DIO_write test %d Passed\n", writeTestCounter);
+		fflush(stdout);
+	}else
+	{
+		printf("DIO_write test %d failed\n", writeTestCounter);
+		fflush(stdout);
+	}
+
+	writeTestCounter++;
+}
+
+void DIO_readTesting(uint8_t gpio, uint8_t bits, uint8_t* value, uint8_t expected_output)
+{
+	static int readTestCounter = 1;
+	uint8_t output;
+
+	output = DIO_Read(gpio, bits, value);
+
+	if(output == expected_output)
+	{
+		printf("DIO_read test %d Passed\n", readTestCounter);
+		fflush(stdout);
+	}else
+	{
+		printf("DIO_read test %d failed\n", readTestCounter);
+		fflush(stdout);
+	}
+
+	readTestCounter++;
+}
+
+void DIO_toggleTesting(uint8_t gpio, uint8_t bits, uint8_t expected_output)
+{
+	static int toggleTestCounter = 1;
+	uint8_t output;
+
+	output = DIO_Toggle(gpio, bits);
+
+	if(output == expected_output)
+	{
+		printf("DIO_toggle test %d Passed\n", toggleTestCounter);
+		fflush(stdout);
+	}else
+	{
+		printf("DIO_toggle test %d failed\n", toggleTestCounter);
+		fflush(stdout);
+	}
+
+	toggleTestCounter++;
+}
+
+int main(void)
+{
+	uint8_t readVariable = 0;
+
+	// DIO_init:
+	// DIO_init test1:
+	DIO_initTesting(0, 0, 0x00, E_OK);

```

Comment:
You may make a struct named test case and fill it with all the data you need.
test data, expected result, actual result, and state.